### PR TITLE
Get valgrind working with python

### LINKED ISSUE #22623: Get valgrind testing working with python
## Reason
We currently don't test valgrind with python, which means we're missing the running of the testers that have specific python dependencies. See https://civet.inl.gov/event/100988/ for valgrind _with_ python.

## Design
Set valgrind requirements such that we can test valgrind with python packages.

## Impact
More testing.

### PR DESCRIPTION
**Pull request by loganharbour**
Closes https://github.com/idaholab/moose/issues/22623

The valgrind recipes have been manually modified to add the python module for just this PR. Once this is in, we'll change them to include python by default.

### CODE REVIEW

File: modules/stochastic_tools/test/tests/samplers/latin_hypercube/tests [hunk 1]
```diff
@@ -75,6 +75,7 @@
     command = "./latin_hypercube.py"
     check_files = latin_hypercube.pdf
     required_python_packages = 'pandas numpy matplotlib'
+    valgrind = NONE
```
Comment (GiudGiud) [inline modules/stochastic_tools/test/tests/samplers/latin_hypercube/tests:78]: what s the concern here

File: modules/heat_conduction/tutorials/introduction/tests [hunk 2]
```diff
@@ -42,6 +42,7 @@
       command = 'therm_soln_compare.py'
       imagediff = 'therm_soln_compare.png'
       required_python_packages = 'pandas matplotlib'
+      valgrind = none
```
Comment (GiudGiud) [inline modules/heat_conduction/tutorials/introduction/tests:45]: what s the concern here? mpl not vlgrind clean?

File: modules/stochastic_tools/test/tests/samplers/latin_hypercube/tests (hunk 1 above)
Comment (loganharbour) [reply modules/stochastic_tools/test/tests/samplers/latin_hypercube/tests:78]: running python doesn't work with valgrind

File: modules/heat_conduction/tutorials/introduction/tests (hunk 2 above)
Comment (loganharbour) [reply modules/heat_conduction/tutorials/introduction/tests:45]: running python doesn't work with valgrind

**Review by GiudGiud — DISMISSED** [review]

**Comment by loganharbour** [comment]
@GiudGiud need another - those ST tests are waaaaay too heavy for even valgrind heavy.

**Review by GiudGiud — APPROVED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**